main.py

```diff
@@ -256,3 +255,40 @@ def kcidb_send_notification(data, context):
         smtp.quit()
     # Acknowledge notification as sent
     SPOOL_CLIENT.ack(notification_id)
+
+
+def kcidb_pick_notification():
+    """
+    Pick abandoned notifications and resend them
+    """
+    for notification_id in SPOOL_CLIENT.unpicked():
+        # Pick abandoned notification and resend
+        message = SPOOL_CLIENT.pick(notification_id)
+        if not message:
+            continue
+        # Send
+        # Set From address, if specified
+        if SMTP_FROM_ADDR:
+            message['From'] = SMTP_FROM_ADDR
+        # Add extra CC, if specified
+        if EXTRA_CC:
+            cc_addrs = message["CC"]
+            if cc_addrs:
+                message.replace_header("CC", cc_addrs + ", " + EXTRA_CC)
+            else:
+                message["CC"] = EXTRA_CC
+        # Connect to the SMTP server
+        smtp = smtplib.SMTP(host=SMTP_HOST, port=SMTP_PORT)
+        smtp.ehlo()
+        smtp.starttls()
+        smtp.ehlo()
+        smtp.login(SMTP_USER, SMTP_PASSWORD)
```

Comment:
You'll have to take care of specifying the correct `SMTP_USER` and `SMTP_PASSWORD`, if you want to actually send mail and test that this works. That is provided you're also using GMail.